pkg/dynamicinstrumentation/diconfig/dwarf.go

```diff
@@ -181,9 +184,6 @@ entryLoop:
 }
 
 func loadDWARF(binaryPath string) (*dwarf.Data, error) {
-	if dwarfData, ok := dwarfMap[binaryPath]; ok {
-		return dwarfData, nil
-	}
 	elfFile, err := safeelf.Open(binaryPath)
```

Comment:
I think `elfFile` needs to closed, but I'm guessing `.DWARF()` won't work afterwards. You may need to slightly refactor this code to allow this.

Reply:
Since we call `.DWARF()` right after it seems it's completely fine to close the file. I've amended the code accordingly.